### PR TITLE
Implemented IQueryable extension methods PagedResult and PagedResultAsync for paginated queries

### PR DESCRIPTION
**Pull request by mysticmind**
Implemented IQueryable extension methods PagedResult and PagedResultAsync for paginated queries. Refer #480.

### CODE REVIEW

File: src/Marten.Testing/Pagination/invoking_pagedquery_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            Func<IQueryable<Target>, IQueryable<Target>> order = null;
+            order = q => q.OrderBy(m => m.Date);
+
+            var result = theSession.Query<Target>().PagedQuery(pageIndex, pageSize, order);
+
+            result[0].Date.ShouldBeLessThan(result[result.Count - 1].Date);
+        }
+
+        [Fact]
+        public void can_return_paged_result_async()
```
Comment (jeremydmiller) [inline src/Marten.Testing/Pagination/invoking_pagedquery_Tests.cs]: Can you use async Task tests instead of relying on GetAwaiter().GetResult() here?

**Comment by jeremydmiller** [comment]
I'm good with this other that the tests. I'd prefer that you didn't bother with all the `this.[prop name]`, but I don't really care that much. I take the attitude of "try not to discourage folks from contributing" in these cases.

I might open a separate issue with this afterward. What you're doing won't work OOTB for compiled queries -- and that's way, way too much work to do over vacation, so January;)

**Comment by mysticmind** [comment]
Thanks for reviewing the PR. I have removed usage of this.[prop name] in code and removed usage of GetAwaiter().GetResult() in unit tests and used async Task tests as per your review comments.

**Comment by jeremydmiller** [comment]
@mysticmind I *think* that I'd vote to merge the Metadata class and logic into the PagedList so that everything is easily available and discoverable to users.

That being just a preference thing, the only real problem is that I think it needs a bit more testing. I think you need more tests on fetching a page in the middle, the "has next page" / "has previous page", and the page counts.

**Comment by mysticmind** [comment]
@jeremydmiller I have taken your suggestions, re-looked at the implementation and refactored the whole feature with better abstractions and unit tests covering all functionality. I am quite happy on how the implementation has evolved to the current form. Please check and let me know your thoughts.

**Pull request merged**